src/main/java/org/folio/dcb/service/impl/ServicePointServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package org.folio.dcb.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.client.feign.InventoryServicePointClient;
+import org.folio.dcb.domain.dto.DcbPickup;
+import org.folio.dcb.domain.dto.HoldShelfExpiryPeriod;
+import org.folio.dcb.domain.dto.ServicePointRequest;
+import org.folio.dcb.exception.ResourceAlreadyExistException;
+import org.folio.dcb.service.ServicePointService;
+import org.springframework.stereotype.Service;
+
+@Service
+@Log4j2
+@RequiredArgsConstructor
+public class ServicePointServiceImpl implements ServicePointService {
+
+  private final InventoryServicePointClient servicePointClient;
+
+  @Override
+  public ServicePointRequest createServicePoint(DcbPickup pickupServicePoint) {
+    log.debug("createServicePoint:: automate service point creation {} ", pickupServicePoint);
+
+    ServicePointRequest servicePointRequest = createServicePointRequest(
+      pickupServicePoint.getServicePointId(),
+      getServicePointName(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName()),
+      getServicePointCode(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName()),
+      getServicePointName(pickupServicePoint.getLibraryName(), pickupServicePoint.getServicePointName())
```

Comment:
For name and discoveryDisplayname of createServicePointRequest, looks like we are using same value. Do we really need to pass the same value twice?

Reply:
It's mandatory field, I can't leave it empty